sequencing-server/src/app.ts

```diff
@@ -134,27 +153,53 @@ app.get('/health', (_: Request, res: Response) => {
 
 app.post('/put-dictionary', async (req, res, next) => {
   const dictionary = req.body.input.dictionary as string;
+  const type = req.body.input.type as string;
   logger.info(`Dictionary received`);
 
-  // Note we ignore comments when parsing dictionary because the parser breaks otherwise.
-  const parsedDictionary = ampcs.parse(dictionary, null, { ignoreComment: true });
+  let parsedDictionary: CommandDictionary | ChannelDictionary | ParameterDictionary;
+  let dictionaryPath: string = '';
+  switch (type) {
+    case DictionaryType.COMMAND: {
+      parsedDictionary = ampcs.parse(dictionary, null, { ignoreComment: true });
+      break;
+    }
+    case DictionaryType.CHANNEL: {
+      parsedDictionary = ampcs.parseChannelDictionary(dictionary);
+      break;
+    }
+    case DictionaryType.PARAMETER: {
+      parsedDictionary = ampcs.parseParameterDictionary(dictionary);
+      break;
+    }
+    default:
+      throw new Error(`POST /dictionary: Unsupported dictionary type: ${type}`);
+  }
+
   logger.info(
-    `Dictionary parsed - version: ${parsedDictionary.header.version}, mission: ${parsedDictionary.header.mission_name}`,
+    `dictionary parsed - version: ${parsedDictionary.header.version}, mission: ${parsedDictionary.header.mission_name}`,
   );
-
-  const commandDictionaryPath = await processDictionary(parsedDictionary);
-  logger.info(`command-lib generated - path: ${commandDictionaryPath}`);
-
+  dictionaryPath = await processDictionary(parsedDictionary, type);
+  logger.info(`lib generated - path: ${dictionaryPath}`);
+
+  let db_name = 'command_dictionary';
```

Comment:
Minor, nonblocking: `table_name` or `db_table_name` are more accurate options.

Reply:
Already had the file (app.ts) open so I made the change to `db_table_name` :P